cmd/juju/interact/pollster.go

```diff
@@ -0,0 +1,515 @@
+		return p.selectOne(schema)
+	}
+	var a string
+	var err error
+	switch schema.Format {
+	case "":
+		// anything
+		a, err = p.Enter(schema.Singular)
+	case jsonschema.FormatURI:
+		a, err = p.EnterVerify(schema.Singular, uriVerify)
+	default:
+		// TODO(natefinch): support more formats
+		return nil, errors.Errorf("unsupported format type: %q", schema.Format)
+	}
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return convert(a, schema.Type[0])
+}
+
+func uriVerify(s string) (ok bool, errMsg string, err error) {
+	if s == "" {
+		return false, "", nil
+	}
+	_, err = url.Parse(s)
+	if err != nil {
+		return false, fmt.Sprintf("Invalid URI: %q", s), nil
+	}
+	return true, "", nil
+}
+
+func supportedArraySchema(schema *jsonschema.Schema) bool {
+	// TODO(natefinch): support arrays without schemas.
+	// TODO(natefinch): support arrays with multiple schemas.
+	// TODO(natefinch): support arrays without Enums.
+	if schema.Items == nil ||
+		len(schema.Items.Schemas) != 1 ||
+		len(schema.Items.Schemas[0].Enum) == 0 ||
+		len(schema.Items.Schemas[0].Type) != 1 {
+		return false
+	}
+	switch schema.Items.Schemas[0].Type[0] {
+	case jsonschema.IntegerType,
+		jsonschema.StringType,
+		jsonschema.BooleanType,
+		jsonschema.NumberType:
+		return true
+	default:
+		return false
+	}
+}
+
+func optFromEnum(schema *jsonschema.Schema) []string {
+	ret := make([]string, len(schema.Enum))
+	for i := range schema.Enum {
+		ret[i] = fmt.Sprint(schema.Enum[i])
+	}
+	return ret
+}
+
+func (p *Pollster) selectOne(schema *jsonschema.Schema) (interface{}, error) {
+	options := make([]string, len(schema.Enum))
+	for i := range schema.Enum {
+		options[i] = fmt.Sprint(schema.Enum[i])
+	}
+	def := ""
+	if schema.Default != nil {
+		def = fmt.Sprint(schema.Default)
+	}
+	a, err := p.Select(List{
+		Singular: schema.Singular,
+		Plural:   schema.Plural,
+		Options:  options,
+		Default:  def,
+	})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if schema.Default != nil && a == "" {
+		return schema.Default, nil
+	}
+	return convert(a, schema.Type[0])
+}
+
+func isObject(schema *jsonschema.Schema) bool {
+	for _, t := range schema.Type {
+		if t == jsonschema.ObjectType {
+			return true
+		}
+	}
+	return false
+}
+
+func convArray(vals []string, schema *jsonschema.Schema) (interface{}, error) {
+	ret := make([]interface{}, len(vals))
+	for i := range vals {
+		v, err := convert(vals[i], schema.Type[0])
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		ret[i] = v
+	}
+	return ret, nil
+}
+
+// convert converts the given string to a specific value based on the schema
+// type that validated it.
+func convert(s string, t jsonschema.Type) (interface{}, error) {
+	switch t {
+	case jsonschema.IntegerType:
+		return strconv.Atoi(s)
+	case jsonschema.NumberType:
+		return strconv.ParseFloat(s, 64)
+	case jsonschema.StringType:
+		return s, nil
+	case jsonschema.BooleanType:
+		switch strings.ToLower(s) {
+		case "y", "yes", "true", "t":
```

Comment:
I actually intentionally did not support 0 or 1, because I don't want anyone writing a prompt that requires someone to know that 0 is true and 1 is false.  I amost didn't support true and false, but those are at least english words with specific meanings.